packages/api/src/webhooks/cannon.ts

```diff
@@ -546,3 +546,13 @@ export default class WebhookCannon {
     );
   }
 }
+
+export async function addSignatureHeader(
+  params: any,
+  webhook: DBWebhook,
+  timestamp: number
+) {
+  let signature = sign(params.body, webhook.sharedSecret);
+  params.headers[SIGNATURE_HEADER] = `t=${timestamp},v1=${signature}`;
+  return params;
+}
```

Comment:
**niiit**: I think this function receives and returns more than it needs to, which makes it harder to know what it does from the call-site. The interface issues I see:
 - it has no reason to be async
 - it could even change the request timeout if it wanted to lol, it gets the full request params
 - one also can't tell what information it needs. looks like it needs the whole webhook spec and request params when it only needs 1 field from each 
 - it looks like it returns a new params object, as if it were thread-safe for example, but it actually mutates the object in-place and then always returns the same thing. callers should never need to re-assign their local variables but every caller does because of that Also misleading IMO.

WDYT of?
```suggestion
export function signatureHeaders(
  payload: any,
  sharedSecret: string,
  timestamp: number
) {
  if (!sharedSecret) return {};

  let signature = sign(payload, sharedSecret);
  return {[SIGNATURE_HEADER]: `t=${timestamp},v1=${signature}`};
}
```
added a suggestion on the call-site to show how it'd look like.